modules/internal/ChapelBase.chpl

```diff
@@ -997,14 +1007,20 @@ module ChapelBase {
     }
   }
 
+  pragma "fn synchronization free"
+  extern proc chpl_comm_task_end(): void;
```

Comment:
I'm not sure it's true that chpl_comm_task_end doesn't have synchronization. That said I'm not quite sure what exactly this pragma means by "synchronization"

Reply:
Offline I think we decided to use "fn synchronization free" for functions that clearly have no "fn synchronization free" and use that as a common flag for optimizations, and we'll also add other flags for things like  "this function is safe for the unordered forall opt" or whatever.

Reply:
I'm changing it to "creates task" "joins task" and the other properties